src/tests/src/libs/antares/study/series/timeseries-tests.cpp

```diff
@@ -0,0 +1,138 @@
+#define BOOST_TEST_MODULE "test time series"
+#define BOOST_TEST_DYN_LINK
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <boost/test/unit_test.hpp>
+
+#include <include/antares/series/series.h>
+
+using namespace Antares::Data;
+
+// =================
+// The fixture
+// =================
+struct Fixture
+{
+    Fixture(const Fixture& f) = delete;
+    Fixture(const Fixture&& f) = delete;
+    Fixture& operator=(const Fixture& f) = delete;
+    Fixture& operator=(const Fixture&& f) = delete;
+    Fixture() : ts(tsnum)
+    {
+        ts.resize(1, HOURS_PER_YEAR);
+        tsnum.resize(1, 1);
+    }
+    TimeSeries ts;
+    TimeSeries::TSNumbers tsnum;
+    std::string folder;
+
+    void fillColumn(unsigned int idx);
+    void fillColumnReverse(unsigned int idx);
+
+    void fillTsnum();
+
+};
+
+void Fixture::fillColumn(unsigned int idx)
+{
+    for (unsigned int i = 0; i < ts.timeSeries.height; i++)
+        ts.timeSeries[idx][i] = i;
+}
+
+void Fixture::fillColumnReverse(unsigned int idx)
+{
+    for (unsigned int i = 0; i < ts.timeSeries.height; i++)
+        ts.timeSeries[idx][i] = HOURS_PER_YEAR - i;
+}
+
+void Fixture::fillTsnum()
+{
+    tsnum.resize(1, ts.timeSeries.width);
+    for (unsigned int i = 0; i < ts.timeSeries.width; i++)
+        tsnum[0][i] = i;
+}
+
+// ==================
+// Tests section
+// ==================
+
+// Here, we need the "lightweight fixture"
+BOOST_AUTO_TEST_SUITE(timeseries_tests)
+
+BOOST_FIXTURE_TEST_CASE(getSeriesIndex, Fixture)
+{
+    tsnum.resize(1, 10);
+    for (unsigned int i = 0; i < 10; i++)
+        tsnum[0][i] = i;
+
+    //timeSeries.width == 1 so returns 0
+    BOOST_CHECK_EQUAL(ts.getSeriesIndex(5), 0);
+
+    ts.resize(2, HOURS_PER_YEAR);
+    for (unsigned int i = 0; i < 10; i++)
+        BOOST_CHECK_EQUAL(ts.getSeriesIndex(i), i);
+}
+
+BOOST_FIXTURE_TEST_CASE(getCoefficientWidth1, Fixture)
+{
+    fillColumn(0);
+    BOOST_CHECK_EQUAL(ts.getCoefficient(0, 12), 12);
+    BOOST_CHECK_EQUAL(ts.getCoefficient(0, 8750), 8750);
+}
+
+BOOST_FIXTURE_TEST_CASE(getCoefficientWidth0, Fixture)
+{
+    ts.resize(0, HOURS_PER_YEAR);
+    BOOST_CHECK_EQUAL(ts.getCoefficient(0, 12), 0);
+    BOOST_CHECK_EQUAL(ts.getCoefficient(0, 8750), 0);
+}
+
+BOOST_FIXTURE_TEST_CASE(getCoefficientNotInitialized, Fixture)
+{
+    ts.resize(4, HOURS_PER_YEAR);
+    fillTsnum();
+    BOOST_CHECK_EQUAL(ts.getCoefficient(3, 12), 0);
+    BOOST_CHECK_EQUAL(ts.getCoefficient(3, 8750), 0);
+}
+
+BOOST_FIXTURE_TEST_CASE(getCoefficientWidthMoreThan1, Fixture)
+{
+    ts.resize(5, HOURS_PER_YEAR);
+    fillTsnum();
+    fillColumn(3);
```

Comment:
This is bad test data too:
we need to have different values in the different columns, otherwise we don't test that we indeed fetch data from the right column.

Reply:
I misunderstood the code: indeed the data is different in other columns (zeros).